lib/ecto/repo/model.ex

```diff
@@ -182,9 +196,47 @@ defmodule Ecto.Repo.Model do
     put_in model.__meta__.state, :loaded
   end
 
-  defp merge_into_changeset(struct, fields, changeset) do
-    changes = Map.take(struct, fields)
-    update_in changeset.changes, &Map.merge(changes, &1)
+  defp sanitize_insert_changeset(struct, fields, embeds, changeset) do
+    old_changes = changeset.changes
```

Comment:
Let's call it `changeset_changes` or just `changes`. Current name does not apply.